packages/react-dev-utils/openBrowser.js

```diff
@@ -8,22 +8,61 @@
  */
 
 var execSync = require('child_process').execSync;
+var spawn = require('cross-spawn');
 var opn = require('opn');
 
 // https://github.com/sindresorhus/opn#app
 var OSX_CHROME = 'google chrome';
 
-function openBrowser(url) {
+
+/**
+ * Use an enum for the possible uses of the BROWSER environment variable
+ * 'browser': open a browser specified by the user. if value is not specified,
+ * use the system's default browser.
+ * 'browser': open a browser.
+ * 'none': Don't open anything.
+ * 'script': run a node.js script.
+ *
+ */
+const BROWSER_TYPES = Object.freeze({
+  'browser': 0,
+  'none': 1,
+  'script': 2,
+})
+
+/**
+ * Returns an object with the BROWSER environment variable's value and type
+ */
+function getBrowserEnv() {
   // Attempt to honor this environment variable.
   // It is specific to the operating system.
   // See https://github.com/sindresorhus/opn#app for documentation.
-  const browser = process.env.BROWSER;
+  const browserStr = process.env.BROWSER;
+  const result = { value: browserStr }
+  if (!browserStr)
+    result.type = BROWSER_TYPES.browser;
+  else if (browserStr && browserStr.endsWith('.js'))
+    result.type = BROWSER_TYPES.script;
+  else if (browserStr.toLowerCase() === 'none')
+    result.type = BROWSER_TYPES.none;
+  else
+    result.type = BROWSER_TYPES.browser;
+  return result
+}
 
-  // Special case: BROWSER="none" will prevent opening completely.
-  if (browser === 'none') {
-    return false;
-  }
+/**
+ * spawns a new child process to execute a node.js script
+ */
+function executeNodeScript(scriptPath, url) {
+  const extraArgs = process.argv.slice(2);
+  const args = [scriptPath]
+                .concat(extraArgs) // add any extra flags
+                .concat(url); // add url last
+  spawn('node', args);
+  return true;
+}
 
+function _openBrowser(browser, url) {
```

Comment:
IMO this naming is confusing. Can we change the names of these functions to reflect what they do exactly instead of using an underscore?